drivers/serial/uart_rpi_pico.c

```diff
@@ -59,13 +60,51 @@ static void uart_rpi_poll_out(const struct device *dev, unsigned char c)
 	uart_hw->dr = c;
 }
 
+static int uart_rpi_set_baudrate(const struct device *dev, uint32_t input_baudrate,
+				 uint32_t *output_baudrate)
+{
+	const struct uart_rpi_config *cfg = dev->config;
+	uart_hw_t * const uart_hw = cfg->uart_regs;
+	uint32_t baudrate_frac;
+	uint32_t baudrate_int;
+	uint32_t baudrate_div;
+	uint32_t pclk;
+	int ret;
+
+	if (input_baudrate == 0) {
+		return -EINVAL;
+	}
+
+	ret = clock_control_get_rate(cfg->clk_dev, (clock_control_subsys_t)cfg->clk_id, &pclk);
+	if (ret < 0 || pclk == 0) {
+		return -EINVAL;
+	}
+
+	baudrate_div = (8 * pclk / input_baudrate);
+	baudrate_int = baudrate_div >> 7;
+	baudrate_frac = baudrate_int == 0 || baudrate_int >= 65535 ? 0 :
+		((baudrate_div & 0x7f) + 1) / 2;
+	baudrate_int = baudrate_int == 0 ? 1 :
+		baudrate_int >= 65535 ? 65535 : baudrate_int;
+
+	uart_hw->ibrd = baudrate_int;
+	uart_hw->fbrd = baudrate_frac;
+
+	uart_hw->lcr_h |= 0;
```

Comment:
Why is this needed?

Reply:
PL011 needs a dummy line control register write to latch in the divisors

Reply:
So please add a comment explaining it